### PR TITLE
Replace the MainDashboardButton slot with a setting in the site editor

### PR DESCRIPTION
**Pull request by youknowriad**
## What?

In #44770 we're redesigning the sidebar of the site editor and it highlighted that the `__experimentalMainDashboardButton` slot was not a semantic API that adapts to the design changes. So I'm replacing it with a setting (dashboardUrl) to allow potential third-party devs to actually replace the back URL.

### CODE REVIEW

File: packages/edit-site/src/components/navigation-sidebar/navigation-panel/index.js [hunk 1]
```diff
@@ -28,7 +28,6 @@ import {
  * Internal dependencies
  */
 import { useLink } from '../../routes/link';
-import MainDashboardButton from '../../main-dashboard-button';
```
Comment (jorgefilipecosta) [inline packages/edit-site/src/components/navigation-sidebar/navigation-panel/index.js:31]: Should we do the same change to edit-post in order to be consistent?
e.g.: remove packages/edit-post/src/components/header/main-dashboard-button/index.js and its docs
docs/reference-guides/slotfills/main-dashboard-button.md?

Comment (youknowriad) [reply packages/edit-site/src/components/navigation-sidebar/navigation-panel/index.js:31]: The post editor one is another story. It's used differently. Some people replace the whole component with their own, so I want to be more careful there.

**Review by jorgefilipecosta — APPROVED** [review]
LGTM 👍

**Comment by youknowriad** [comment]
I've checked wpdirectory and this is not used at all.

**Pull request merged**